internal/core/key.go

```diff
@@ -626,6 +626,25 @@ func (k *IndexDataStoreKey) ToString() string {
 	return string(k.Bytes())
 }
 
+// IsEqual returns true if the two keys are equal
+func (k *IndexDataStoreKey) IsEqual(other IndexDataStoreKey) bool {
```

Comment:
nitpick: See nitpick in `client` package about `IsEqual` vs `Equal`, if you change those, please change this.

Reply:
done